doc/version_history.rst

```diff
@@ -2,6 +2,23 @@
 Version History
 ===============
 
+v0.4.1
+------
+* In `CSC`:
+  * send ack_in_progress in `begin_` methods of state transition commands, since they are called before the state transition.
+  * send ack_in_progress for all CSC commands that use timeout information.
+  * In `_telemetry_loop`, refactor how to get new messages and add information about message consumption rate. If queue is not empty, get with `get_nowait` otherwise use asynchronous method. This will cause the loop to pause and wait for new messages to arrive asynchronously, without the need to pool for new data while at the same time, reading as fast as possible when the queue is not empty.
```

Comment:
The penalty for not using empty() is about 5%. Shall be added to description? Proof:

```
import asyncio


class Test:
    def __init__(self):
        self.q = asyncio.Queue()
        self.num = 1000000

    async def queuein(self):
        for i in range(self.num):
            await self.q.put(i)

    async def no_wait(self):
        for i in range(self.num):
            m = await self.q.get() if self.q.empty() else self.q.get_nowait()

    async def get(self):
        for i in range(self.num):
            m = await self.q.get()

    async def run(self, get):
        producer = asyncio.create_task(self.queuein())
        if get:
             consumer = asyncio.create_task(self.get())
        else:
             consumer = asyncio.create_task(self.no_wait())

        await producer
        await consumer
```

```
In [2]: import tt

In [3]: a=tt.Test()

In [5]: import asyncio

In [6]: timeit asyncio.run(a.run(False))
2.11 s ± 105 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [7]: timeit asyncio.run(a.run(True))
2.22 s ± 30 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [8]: timeit asyncio.run(a.run(True))

2.2 s ± 14 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [9]: timeit asyncio.run(a.run(False))
2.07 s ± 25.2 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```